deployment/terraform/live/main.tf

```diff
@@ -15,7 +15,7 @@ provider "aws" {
 module "object_storage" {
   source           = "../modules/s3"
   name             = "${terraform.workspace} object_storage"
-  bucket_name_base = "${terraform.workspace}"
+  bucket_name_base = "${replace(terraform.workspace, "/[^A-Za-z0-9]/", "-")}"
```

Comment:
Illegal names should produce an error and not be silently converted.

Reply:
I've hunted for a little bit, but I'm not finding a way to constrain user-supplied variables with terraform. If we remove this and just let the error pop up downstream, it's a little hard to debug: it doesn't echo the value which fails, or tell you where it came from. Also, different parts of AWS have different constraints around naming: Forcing the least common denominator across the board is not necessary.

Reply:
This expression does not cover all the cases anyway. It is much easier and more informative to describe these constraints for the workspace name in the docs, or if `bucket_name_base` is an input variable then in its `description` parameter.